ui/src/errors.ts

```diff
@@ -6,21 +6,37 @@
 import { localize } from "./localize";
 
 // tslint:disable: max-classes-per-file
+export class AzureToolsError extends Error {
+    public suppressDisplay: boolean = false;
+    public suppressReportIssue: boolean = false;
```

Comment:
Instead of individual properties like `suppressReportIssue`, what if we let them specify a `Partial<IActionContext>` so that they can overwrite anything related to error handling _or_ telemetry (Stephen in particular has wanted to specify additional telemetry properties from errors for a while now haha). It could even be called `OverwriteActionContextError` or something

Reply:
Yes, that would be more open. Regarding name, how about ErrorWithActionContext. I don't want to use "Overwrite", because in error handling, we may not always use the value from this error, sometimes the action might define some behavior which is what we need to use.

Reply:
Yeah I was thinking we could do a `Partial<IActionContext>` or it could be a callback/method that takes in the current `IActionContext`. Not sure if people will want to edit the existing or just specify things to overwrite